.github/workflows/lib_compat_test.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+# Verifies a binary uses only dynamic symbols from whitelisted library versions.
+# Prints the disallowed symbols and their versions on failure.
+# Usage: lib_compat_test.py bin/clangd --lib=GLIBC_2.18
+
+import argparse
+import subprocess
+import sys
+
+parser = argparse.ArgumentParser()
+parser.add_argument('binary')
+parser.add_argument('--lib', action='append', default=[],
+                    help="Whitelist a library, e.g. GLIBC_2.18 or GLIBC")
+parser.add_argument('--sym', action='append', default=[],
+                    help="Whitelist a symbol, e.g. crc32")
+args = parser.parse_args()
+
+# Parses GLIBC_2.3 into ("GLIBC", [2,3])
+# Parses GLIBC into ("GLIBC", None)
+def parse_version(version):
```

Comment:
do we expect this to parse anything but `GLIBC` ? do we now if other libraries also use `_` as a separator between version and nodename ?

maybe make it specific to `GLIBC` for now and workout later if need be?

Reply:
This is really useful for poking at the effect of build configs (forgot to put it in the patch description).
My normal (non-special-portable flags) clangd build has GLIBC, GLIBCXX (which is libstdc++), ZLIB, NCURSES6_TINFO, CXXABI, GCC - they all use the versioning scheme.

Reply:
Yeah I wouldn't know what the semantics of that was, so having the script blow up or fail is ideal.